### PR TITLE
Fixing option encoding through unset

### PR DESCRIPTION
**Pull request by LMnet**
Continuing of my [previous pull request](https://github.com/ringcentral/cassandra4io/pull/24) about `unset`.

I practically revert my previous change: by default `Option` binder will again use `setToNull`. To use `unset` an `usingUnset` extension method should be explicitly called inside an interpolator string. 

What else have done:
* `BoundValue` is now public. Without this `usingUnset` will not work.
* More tests about `null` and `usingUnset`.
* Docs update.
* Small additional improvement: `CqlStringContext` is now `AnyVal`. To get rid of extra allocations.

### CODE REVIEW

File: src/main/scala/com/ringcentral/cassandra4io/cql/package.scala [hunk 1]
```diff
@@ -214,10 +214,15 @@ package object cql {
...
-    val cql      = new CqlStringInterpolator(ctx)
-    val cqlConst = new CqlConstInterpolator(ctx)
+  implicit class CqlStringContext(val ctx: StringContext) extends AnyVal {
+    def cqlt     = new CqlTemplateStringInterpolator(ctx)
+    def cql      = new CqlStringInterpolator(ctx)
+    def cqlConst = new CqlConstInterpolator(ctx)
+  }
+
+  implicit class UnsetOptionValueOps[A](self: Option[A]) {
+    def usingUnset(implicit aBinder: Binder[A]): BoundValue[Option[A]] =
```
Comment (myazinn) [inline src/main/scala/com/ringcentral/cassandra4io/cql/package.scala:224]: How about `.ignoreNone` / `.ignoreNull` ? Not that it really important :)

Comment (LMnet) [reply src/main/scala/com/ringcentral/cassandra4io/cql/package.scala:224]: `ignoreNone`/`ignoreNull` introduces some new naming, which is completely unrelated to an underlying mechanism (`unset`). I believe that something with `unset` will be more familiar for Cassandra users.

I also think about `withUnset`, which is fine too.

**Review by Alexey-Yuferov — APPROVED** [review]
